lib/tools/system-calls.js

```diff
@@ -855,25 +845,43 @@ systemCallMethods.reboot = async function reboot (retries = DEFAULT_ADB_REBOOT_R
   });
 };
 
+/**
+ * @typedef {Object} rootResult
+ * @property {boolean} isRooted True if the device is rooted now
+ * @property {boolean} wasAlreadyRooted True if the device was already rooted
+ */
+
 /**
  * Switch adb server to root mode.
- *
- * @return {boolean} True of the switch was successful or false
- *                   if the switch failed.
+ * @param {boolean} unroot - Should we unroot instead of root? (default false)
+ * @return {boolean} True if the device is rooted.
  */
-systemCallMethods.root = async function root () {
+systemCallMethods.root = async function root (unroot = false) {
+  let cmd = unroot ? 'unroot' : 'root';
+  let wasAlreadyRooted = false;
   try {
-    let {stdout} = await exec(this.executable.path, ['root']);
+    let {stdout} = await exec(this.executable.path, [cmd]);
 
     // on real devices in some situations we get an error in the stdout
-    if (stdout && stdout.indexOf('adbd cannot run as root') !== -1) {
-      throw new Error(stdout.trim());
+    if (stdout) {
+      if (stdout.includes('adbd cannot run as root')) {
```

Comment:
if NOT includes

Reply:
This is what it was before:

```js
if (stdout && stdout.indexOf('adbd cannot run as root') !== -1) {
  throw new Error(stdout.trim());
}
```

`stdout.indexOf('adbd cannot run as root') !== -1` is equivalent to `stdout.includes('adbd cannot run as root')`

I think we want to throw the error if we get that message

Reply:
I think it would be server to remove this exception at all and only log a warning